plugins/backstage-plugin-flux/src/components/helpers.tsx

```diff
@@ -122,30 +123,44 @@ export function SuspendButton({
   toggleSuspend,
   status,
   readOnly,
+  suspendMessage,
+  setSuspendMessage,
 }: {
   resource: Source | Deployment;
   toggleSuspend: () => Promise<void>;
   status: boolean;
   readOnly?: boolean;
+  suspendMessage: string;
+  setSuspendMessage: React.Dispatch<React.SetStateAction<string>>;
 }) {
+  const [suspendMessageModalOpen, setSuspendMessageModalOpen] = useState(false);
   const classes = useStyles();
   const label = `${resource.namespace}/${resource.name}`;
   const title = status ? `Suspending ${label}` : `Suspend ${label}`;
 
   return (
-    <Tooltip title={readOnly ? 'Read-only mode is enabled' : title}>
-      <div>
-        <IconButton
-          data-testid={`suspend ${label}`}
-          className={classes.actionButton}
-          size="small"
-          onClick={toggleSuspend}
-          disabled={resource.suspended || readOnly}
-        >
-          <PauseIcon />
-        </IconButton>
-      </div>
-    </Tooltip>
+    <>
+      <Tooltip title={readOnly ? 'Read-only mode is enabled' : title}>
+        <div>
+          <IconButton
+            data-testid={`suspend ${label}`}
+            className={classes.actionButton}
+            size="small"
+            onClick={() => setSuspendMessageModalOpen(!suspendMessageModalOpen)}
+            disabled={resource.suspended || readOnly}
+          >
+            <PauseIcon />
+          </IconButton>
+        </div>
+      </Tooltip>
+      <SuspendMessageModal
```

Comment:
WDYT about lifting this out of the table / trying to "protect it" with another memo?

It might still get wiped right now (if some other resource legitimately transitions), but I have not repro'd this yet either.